gui/src/3D/controls/slider3D.ts

```diff
@@ -0,0 +1,246 @@
+    }
+
+    /**
+     * Gets the mesh used to render this control
+     */
+    public get mesh(): Nullable<AbstractMesh> {
+        if (this.node) {
+            return this._sliderThumb;
+        }
+
+        return null;
+    }
+
+    /** Gets or sets minimum value */
+    public get minimum(): number {
+        return this._minimum;
+    }
+
+    public set minimum(value: number) {
+        if (this._minimum === value) {
+            return;
+        }
+
+        this._minimum = Math.max(value, SLIDER_MIN);
+        this._value = Math.max(Math.min(this._value, this._maximum), this._minimum);
+    }
+
+    /** Gets or sets maximum value */
+    public get maximum(): number {
+        return this._maximum;
+    }
+
+    public set maximum(value: number) {
+        if (this._maximum === value) {
+            return;
+        }
+
+        this._maximum = Math.max(value, this._minimum);
+        this._value = Math.max(Math.min(this._value, this._maximum), this._minimum);
+    }
+
+    /** Gets or sets step value */
+    public get step(): number {
+        return this._step;
+    }
+
+    public set step(value: number) {
+        if (this._step === value) {
+            return;
+        }
+
+        this._step = Math.max(Math.min(value, this._maximum - this._minimum), SLIDER_STEP);
+    }
+
+    /** Gets or sets current value */
+    public get value(): number {
+        return this._value;
+    }
+
+    public set value(value: number) {
+        if (this._value === value) {
+            return;
+        }
+
+        this._value = Math.max(Math.min(value, this._maximum), this._minimum);
+        this.onValueChangedObservable.notifyObservers(this._value);
+    }
+
+    protected get start(): number {
+        if (!this.node) {
+            return -SLIDER_SCALING / 2;
+        }
+
+        return this._sliderBar.position.x - this._sliderBar.scaling.y / 2;
+    }
+
+    protected get end(): number {
+        if (!this.node) {
+            return SLIDER_SCALING / 2;
+        }
+
+        return this._sliderBar.position.x + this._sliderBar.scaling.y / 2;
+    }
+
+    protected get roundingToDecimal(): boolean {
+        if (!this.mesh) {
+            return false;
+        }
+
+        return this.minimum % 1 !== 0 || this.maximum % 1 !== 0 || this.step !== 0;
+    }
+    /**
+     * Gets the slider bar material used by this control
+     */
+    public get sliderBarMaterial(): StandardMaterial {
+        return this._sliderBarMaterial;
+    }
+
+    /**
+     * Gets the slider thumb material used by this control
+     */
+    public get sliderThumbMaterial(): StandardMaterial {
+        return this._sliderThumbMaterial;
+    }
+
+    // Mesh association
+    protected _createNode(scene: Scene): TransformNode {
+        const anchor = new TransformNode(`${this.name}_slider`, scene);
+
+        const sliderBar = MeshBuilder.CreateCylinder(`${this.name}_sliderbar`, { diameter: 0.03, height: 1.0 }, scene);

```

Comment:
>0.03 [](start = 91, length = 4)

Do we think that an external developer will need to customize these values? or can they be reset after slider control creation?

Reply:
I was assuming the developer can use scaling get/set property to scale the control  up/down to the desired size. I think of these values as scale of 1 for the control as a whole. I don't think we need these to be customizable.  Let me know your thought.

Reply:
I am for allowing customization in this case. Mainly because you are changing the scaling.y value. A user that will manipulate the scaling values as they see fit will have these values overwritten. Might be enough to define a `scaleFactor` that will be applied to all of those default values.

Reply:
The user can use getter/setter public property for scaling the control up or down.  Those default values are like a seed that will give the control the default look.